src/Idris/Package/Init.idr

```diff
@@ -67,25 +68,50 @@ prompt p = putStr p >> fflush stdout >> getLine
 
 export
 covering
-interactive : IO PkgDesc
+interactive : IO (Either () PkgDesc)
 interactive = do
-  pname    <- prompt "Package name: "
-  pauthors <- prompt "Package authors: "
-  poptions <- prompt "Package options: "
-  psource  <- prompt "Source directory: "
-  let sourcedir = mstring psource
-  modules  <- findModules sourcedir
-  let pkg : PkgDesc =
-        { authors   := mstring pauthors
-        , options   := (emptyFC,) <$> mstring poptions
-        , modules   := modules
-        , sourcedir := sourcedir
-        } (initPkgDesc (fromMaybe "project" (mstring pname)))
-  pure pkg
+  pname <- prompt "Package name: "
+  -- check to ensure that pname is a valid Idris2 identifier.
+  case checkPackageName $ fastUnpack pname of
+    False => do () <- putStrLn "Package name is not a valid Idris Identifier."
+                pure $ Left ()
+    True  => do pauthors <- prompt "Package authors: "
+                poptions <- prompt "Package options: "
+                psource  <- prompt "Source directory: "
+                let sourcedir = mstring psource
+                modules  <- findModules sourcedir
+                let pkg : PkgDesc =
+                      { authors   := mstring pauthors
+                      , options   := (emptyFC,) <$> mstring poptions
+                      , modules   := modules
+                      , sourcedir := sourcedir
+                      } (initPkgDesc (fromMaybe "project" (mstring pname)))
+                pure $ Right pkg
 
   where
 
     mstring : String -> Maybe String
     mstring str = case trim str of
       "" => Nothing
       str => Just str
+
+    isIdentStart : Char -> Bool
```

Comment:
I suspect that code (and below) came from the lexer but there is no way to
import and reuse the bits for identifiers, is that right?

Reply:
@andrevidela  I don't think there's a straightforward to use that code in this context (could be wrong), but totally open to changing the way I'm doing this if there is a better way (or a way that can utilize the lexer code).